packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -621,14 +635,18 @@ export class Viewer {
       this.clock = undefined;
 
       this.streamDisposable = await this.connectStream(resource);
-      await this.stream.reconnect({
+      const result = await this.stream.reconnect({
         streamId: { hex: streamId },
         dimensions: this.dimensions,
         frameBackgroundColor: this.getBackgroundColor(),
         streamAttributes: this.getStreamAttributes(),
       });
       this.isStreamStarted = true;
       this.isReconnecting = false;
+
+      this.jwt = result.reconnect?.jwt || undefined;
+      this.jwtRefresh.emit(this.jwt);
```

Comment:
if these are only valid for 5 minutes, are there updates on more messages that aren't start or reconnect?

is it guaranteed it is different than the previous jwt on start and reconnect messages? we wouldn't want to emit the event if it was the same

Reply:
It's a new token on each reconnect. How often do reconnects happen?

Reply:
Reconnects should happen ~ every 1 minutes.

Reply:
See Dan Shultz' comment. I changed the event to be a connectionChanged event and it has a jwt on connect/reconnect events.